vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold.cc

```diff
@@ -276,4 +281,88 @@ void Uphold::CreateAnonAddressIfNecessary(
   card_->CreateAnonAddressIfNecessary(std::move(wallet), callback);
 }
 
+void Uphold::SaveTransferFee(ledger::TransferFeePtr transfer_fee) {
+  if (!transfer_fee) {
+    return;
+  }
+
+  auto timer_id = 0u;
+  SetTimer(&timer_id);
+  transfer_fee->execution_id = timer_id;
+  ledger_->SetTransferFee(ledger::kWalletUphold, std::move(transfer_fee));
+}
+
+void Uphold::OnTransferFeeCompleted(
+    const ledger::Result result,
+    const bool created,
+    const ledger::TransferFee& transfer_fee) {
+  if (result == ledger::Result::LEDGER_OK) {
+    ledger_->RemoveTransferFee(ledger::kWalletUphold, transfer_fee.id);
+    return;
+  }
+
+  SaveTransferFee(ledger::TransferFee::New(transfer_fee));
+}
+
+void Uphold::TransferFee(
+    const ledger::Result result,
+    ledger::ExternalWalletPtr wallet,
+    const ledger::TransferFee& transfer_fee) {
+
+  if (result != ledger::Result::LEDGER_OK) {
+    SaveTransferFee(ledger::TransferFee::New(transfer_fee));
+    return;
+  }
+
+  auto callback = std::bind(&Uphold::OnTransferFeeCompleted,
+          this,
+          _1,
+          _2,
+          transfer_fee);
+
+      transfer_->Start(
+          transfer_fee.amount,
+          GetFeeAddress(),
+          std::move(wallet),
+          callback);
+}
+
+void Uphold::TransferFeeOnTimer(const uint32_t timer_id) {
+  const auto fees = ledger_->GetTransferFees(ledger::kWalletUphold);
+
+  for (auto const& value : fees) {
+    const auto fee = *value.second;
+    if (fee.execution_id == timer_id) {
+      auto callback = std::bind(&Uphold::TransferFee,
+          this,
+          _1,
+          _2,
+          fee);
+
+      ledger_->GetExternalWallet(ledger::kWalletUphold, callback);
+      return;
+    }
+  }
+}
+
+void Uphold::OnTimer(uint32_t timer_id) {
+  BLOG(ledger_, ledger::LogLevel::LOG_INFO)
+    << "OnTimer Uphold: "
+    << timer_id;
+
+  TransferFeeOnTimer(timer_id);
+}
+
+void Uphold::SetTimer(uint32_t* timer_id, uint64_t start_timer_in) {
+  if (start_timer_in == 0) {
+    start_timer_in = brave_base::random::Geometric(45);
```

Comment:
Is 45s arbitrary, or a pre-determined/required value?

Reply:
this is our default for requests that don't have defined delay